include/ck/tensor_operation/gpu/element/element_wise_reduce_operation.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+#include "data_type.hpp"
+
+namespace ck {
+namespace tensor_operation {
+namespace element_wise {
+
+struct ReduceSum
+{
+    __host__ __device__ static constexpr float GetReduceZeroValue() { return float(0); }
```

Comment:
I'd opt for GetInitValue. Because the initial value changes based on actual operation type.

For summation reduction, initial value by default is 0.
For multiplication reduction initial value by default is 1.
For max reduction, the initial value by default is -inf.

Reply:
Conventionally, identity value is a property of the reduce operator, that is, reduce(identity, x) := x, while Initial value is a user defined value that can be passed to the reduce operation.

Reply:
Thanks for clarification. The case still holds: the user calls `GetInitValue()`, while the initial value gets initialized by `GetIdentityValue()` if not supplied by the user. 

```c
struct ReduceOp
{
    float init_;
    static float GetIdentityValue() {
         /* return user defined identity so that reduce(identity, x) == x */ 
    }
    
    ReduceOp(float init = GetIdentityValue()) : init_(init) {}

    float GetInitValue() { return init_; }
};
```

Reply:
I had similar debate with @qianfengz before, but didn't have a conclusion.

https://docs.oracle.com/javase/tutorial/collections/streams/reduction.html

It seems we should use "reduction identity value"

Reply:
I think we all agree that "IdentityValue" should be a property of the type of reduction, it should be a deterministic value that should not be specified by user.

I think "init value" is also needed, in cases like multi-stage reduction.

In my use cases, I don't need "init value".

I will change the name from "GetReductionZeroValue()" to "GetIdentityValue()" for now.

@qianfengz will refactor this part of code later, using his reduction operators, and make both of terminologies formal definition:
1. "IdentityValue"
2. "InitialValue"